src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -265,6 +265,7 @@ public void CloseAppenders()
         /// <param name="expireTime">The time which prior the appenders considered expired</param>
         public void CloseAppenders(DateTime expireTime)
         {
+            lock(this)
```

Comment:
lock(this) is a bad practise, since you lock the entire instance.

cosider:

var appenderCache = xxxx.

lock(appenderCache)
{
Thread.Sleep(TimeSpan.FromHours(5));
}

while inside AppenderCache instance
lock(this)
{
 is blocked 5 hours.
}

Its always better to make a class private lock objects.